assets/src/dashboard/utils/useFocusOut.js

```diff
@@ -29,23 +30,40 @@ function useFocusOut(ref, callback, deps) {
     const onFocusOut = (evt) => {
       // If focus moves somewhere outside the node, callback time!
       if (evt.relatedTarget && !node.contains(evt.relatedTarget)) {
+        isFocused = false;
         callback();
       }
     };
 
+    const onFocusIn = () => {
+      isFocused = true;
+    };
+
     const onDocumentClick = (evt) => {
       // If something outside the target node is clicked, callback time!
       const isInDocument = node.ownerDocument.contains(evt.target);
       const isInNode = node.contains(evt.target);
       if (!isInNode && isInDocument) {
+        isFocused = false;
         callback();
       }
+      // Handle text element editor case
+      const activeElementInNode = node.contains(document.activeElement);
+      if (activeElementInNode) {
+        isFocused = true;
+      }
```

Comment:
Is it only necessary for text element cases? Not sure if we should create specificities like this for hooks that can be reusable across the editor/dashboard. Instead, I would recommend adding the potential fix inside the text element codebase on the Editor.

Reply:
Right now, just this case is "unique" - it's more complex than others because it involves 3P library with auto-focus and it was the cleanest (and working) way to handle it. But it can catch other more complex cases too.